facebook/docusaurus
# feat(v2): add ability default lang for code blocks

## Motivation

It often happens that in Markdown code blocks do not (intentionally) specify the language, and then the default language was used. This is convenient if the examples language in docs is the same (_why should it be indicated all the time?_). Or, for example, when end-user forgot to specify that language.

Note: if you do not specify a language, then in HTML it will be represented as `language-undefined`.

In v1 there was such an feature, I propose to bring back it to v2.

Also in this PR, as an experiment, changed way to provide options for Prism - in a separate object (as it was in [v1](https://docusaurus.io/docs/en/site-config#highlight)). This is a breaking change, yes, but in the end it is a flexible approach, because in the future there may be more options for highlighting code (=Prism).

```diff
- themeConfig.prismTheme
- themeConfig.prismDefaultLanguage
+ themeConfig.prism: {
+   theme: ...
+   defaultLanguage: ...
+   // and so on
+ }
```

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

1. Set the following options in `docusaurus.config.js`:

```js
// ...  
  themeConfig: {
  // ...
    prism: {
      defaultLanguage: 'js',
      theme: require(...),
    },
```

2. In docs, omit the language in the code block:

<pre>
```
console.log('JS is my default language!')
```
</pre>

3. Make sure that the given code block is highlighted as `js`:


```js
console.log('JS is my default language!')
```
